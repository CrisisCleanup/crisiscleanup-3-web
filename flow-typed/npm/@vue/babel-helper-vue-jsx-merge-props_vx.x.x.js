// flow-typed signature: d7a23c53f4091ef2bf1df8afa5eb43c4
// flow-typed version: <<STUB>>/@vue/babel-helper-vue-jsx-merge-props_v^1.0.0/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-helper-vue-jsx-merge-props'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-helper-vue-jsx-merge-props' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper.js' {
  declare module.exports: $Exports<'@vue/babel-helper-vue-jsx-merge-props/dist/helper'>;
}
