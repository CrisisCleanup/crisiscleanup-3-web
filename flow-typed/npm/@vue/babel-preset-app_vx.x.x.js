// flow-typed signature: 6ba91a92845d9a556a5bcdcc4f033fed
// flow-typed version: <<STUB>>/@vue/babel-preset-app_v^4.3.1/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-app'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-app' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-app/polyfillsPlugin' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-app/index' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/index.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/polyfillsPlugin.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app/polyfillsPlugin'>;
}
