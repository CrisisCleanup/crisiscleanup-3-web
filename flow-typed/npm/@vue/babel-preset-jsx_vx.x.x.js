// flow-typed signature: eeecaeff29e4a3a6c63c4fe7e16a0eb3
// flow-typed version: <<STUB>>/@vue/babel-preset-jsx_v^1.1.2/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-jsx'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-jsx' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-jsx/dist/plugin.cjs' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-jsx/dist/plugin.cjs.js' {
  declare module.exports: $Exports<'@vue/babel-preset-jsx/dist/plugin.cjs'>;
}
