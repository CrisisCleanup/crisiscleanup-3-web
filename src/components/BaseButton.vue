<template>
  <button
    :class="[styles, buttonSelector]"
    :alt="alt || text"
    :disabled="disabled || loading"
    type="submit"
    :size="size"
    :title="buttonTitle"
    :data-cy="buttonSelector"
    @click.prevent="performAction"
  >
    <font-awesome-icon
      v-if="loading || showSpinner"
      size="sm"
      icon="spinner"
      spin
    />
    <font-awesome-icon v-if="icon" class="m-1" :icon="icon" :size="iconSize" />
    <ccu-icon v-if="ccuIcon" class="m-1" :type="ccuIcon" :size="iconSize" />
    <slot>{{ text }}</slot>
    <font-awesome-icon v-if="suffixIcon" class="m-1" :icon="suffixIcon" />
  </button>
</template>

<script>
import { kebabCase } from 'lodash';
import VueTypes from 'vue-types';
import { BUTTON_VARIANTS as VARIANTS, ICON_SIZES, ICONS } from '@/constants';
import { EventsMixin } from '@/mixins';
import { delay } from '@/utils/promise';

export default {
  name: 'BaseButton',
  mixins: [EventsMixin],
  props: {
    action: VueTypes.func,
    disabled: VueTypes.bool.def(false),
    showSpinner: VueTypes.bool.def(false),
    text: VueTypes.string,
    alt: VueTypes.string.def('button'),
    title: VueTypes.string.def('Button'),
    size: VueTypes.oneOf(['small', 'medium', 'large', 'sm', 'md', 'lg']),
    icon: VueTypes.string,
    ccuIcon: VueTypes.oneOf(Object.values(ICONS)),
    iconSize: VueTypes.oneOf(ICON_SIZES).def('sm'),
    suffixIcon: VueTypes.oneOf(Object.values(ICONS)),
    selector: VueTypes.string,
    variant: VueTypes.oneOf(Object.values(VARIANTS)),
  },
  data() {
    return {
      loading: false,
    };
  },
  computed: {
    buttonSelector() {
      return this.selector || `js-${kebabCase(this.buttonTitle)}`;
    },
    buttonTitle() {
      return this.text || this.alt || this.title;
    },
    styles() {
      const styles = {
        'large text-h3 font-h3': ['large', 'lg'].includes(this.size),
        'medium text-sans text-h4 font-h4': ['medium', 'md'].includes(
          this.size,
        ),
        'small text-bodysm font-bodysm': ['small', 'sm'].includes(this.size),
        disabled: this.disabled,
        // **** DEPRECATED ****
        primary: this.type === 'primary',
        danger: this.type === 'danger',
        warning: this.type === 'warning',
        link: this.type === 'link',
        bare: this.type === 'bare',
        // **************
        flex: true,
        'items-center': true,
        'justify-center': true,
        'base-button': true,
      };
      if (this.variant) {
        styles[this.variant] = true;
      }
      styles[this.variant] = true;
      return styles;
    },
  },
  methods: {
    async timeout() {
      await delay(5000);
      return Promise.reject();
    },
    async performAction() {
      this.loading = true;

      try {
        if (this.action) {
          await Promise.race([this.action(), this.timeout()]);
        }
      } catch (e) {
        // TODO: expose method for handling button exceptions
      } finally {
        this.logEvent();
        this.loading = false;
      }
    },
  },
};
</script>

<style scoped>
button {
  @apply font-sans;
  cursor: pointer;
  outline: 0;
  transition: all 300ms ease;
}
button:focus {
  outline: 0;
}

/** ----- DEPRECATED ----- */
.primary {
  @apply bg-primary-light;
}
.link {
  @apply text-primary-dark;
}
/** ----- DEPRECATED ----- */

button.solid {
  @apply bg-primary-light;
}

button.solid:hover {
  @apply bg-crisiscleanup-yellow-100;
}

button.solid.disabled {
  @apply bg-crisiscleanup-dark-200 !important;
  @apply text-gray-100;
}

button.outline {
  background-color: transparent;
  border: 1px solid black;
}

button.outline:hover {
  @apply bg-crisiscleanup-dark-500;
  @apply text-gray-100;
}

button.outline.disabled {
  @apply text-crisiscleanup-grey-900;
}

button.outline.disabled:hover {
  @apply text-crisiscleanup-grey-900;
}

button.outline-dark {
  background-color: transparent;
  border: 1px solid white;
}

button.outline-dark:hover {
  @apply bg-crisiscleanup-dark-500;
  @apply text-gray-100;
}

button.outline-dark.disabled {
  @apply text-crisiscleanup-grey-900;
}

button.outline-dark.disabled:hover {
  @apply text-crisiscleanup-grey-900;
}

button.text {
  background-color: transparent;
}

button.text:hover {
  @apply bg-crisiscleanup-grey-100;
}

button.text.disabled {
  @apply text-crisiscleanup-grey-900;
}

button.text.disabled:hover {
  @apply text-crisiscleanup-grey-900;
}

button.small {
  min-height: 24px;
  padding-left: 15px;
  padding-right: 15px;
}

button.medium {
  min-height: 32px;
  padding-left: 20px;
  padding-right: 20px;
}

button.large {
  min-height: 50px;
  padding-left: 34px;
  padding-right: 34px;
}
</style>
