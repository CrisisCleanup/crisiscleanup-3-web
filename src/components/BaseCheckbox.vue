<template>
  <label class="checkbox-container">
    <slot></slot>
    <input
      type="checkbox"
      :checked="value"
      class="checkmark-input"
      @input="$emit('input', $event.target.checked)"
      @change="$emit('change', $event.target.checked)"
    />
    <span class="checkmark"></span>
  </label>
</template>
<script>
export default {
  name: 'BaseCheckbox',
  props: ['value'],
};
</script>

<style scoped>
/* Customize the label (the container) */
.checkbox-container {
  display: block;
  position: relative;
  padding-left: 30px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

/* Hide the browser's default checkbox */
.checkbox-container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

/* Create a custom checkbox */
.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  border: solid 1px #c8cacf;
  background-color: white;
}

/* When the checkbox is checked, add a blue background */
.checkbox-container input:checked ~ .checkmark {
  background-color: #000000;
}

.checkbox-container input:disabled ~ .checkmark {
  background-color: #e4e5e7;
}

/* Create the checkmark/indicator (hidden when not checked) */
.checkmark:after {
  content: '';
  position: absolute;
  display: none;
}

/* Show the checkmark when checked */
.checkbox-container input:checked ~ .checkmark:after {
  display: block;
}

/* Style the checkmark/indicator */
.checkbox-container .checkmark:after {
  left: 7px;
  top: 3px;
  width: 5px;
  height: 10px;
  border: solid white;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}
</style>
