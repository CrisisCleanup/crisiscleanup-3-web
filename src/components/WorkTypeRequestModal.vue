<template>
  <modal
    :title="$t('workTypeRequestModal.work_type_request')"
    modal-classes="w-2/3"
    @close="$emit('onCancel')"
  >
    <div class="px-6 py-3">
      <div class="font-xs my-2">
        <span
          v-html="
            $t('workTypeRequestModal.request_modal_instructions', {
              organizations: organizations.map((org) => org.name).join(','),
              my_organization: myOrganization.name,
              case_number: '',
            })
          "
        ></span>
        <span
          v-html="$t('workTypeRequestModal.please_add_respectful_note')"
        ></span>
        <ul class="my-2 list-disc list-inside">
          <li>
            {{ $t('workTypeRequestModal.reason_member_of_faith_community') }}
          </li>
          <li>
            {{ $t('workTypeRequestModal.reason_working_next_door') }}
          </li>
          <li>
            {{ $t('workTypeRequestModal.reason_we_did_the_work') }}
          </li>
        </ul>
      </div>
      <div class="font-xs my-2 font-bold">
        {{ $t('workTypeRequestModal.contacts') }}
      </div>
      <template v-for="organization in organizations">
        <div v-for="contact in organization.incident_primary_contacts">
          {{ contact.first_name }} {{ contact.last_name }} ({{
            organization.name
          }}) {{ contact.email }} {{ contact.mobile }}
        </div>
      </template>
      <div class="font-xs my-2 font-bold">
        {{ $t('workTypeRequestModal.choose_work_types') }}
      </div>
      <div v-for="work_type_to_request in workTypes">
        <base-checkbox
          class="mb-3"
          :value="requestedWorkTypes.has(work_type_to_request.work_type)"
          @input="
            (value) => {
              setSelectedWorkTypeRequest(work_type_to_request.work_type, value);
            }
          "
        >
          <span class="text-sm">{{
            work_type_to_request.work_type | getWorkTypeName
          }}</span>
        </base-checkbox>
      </div>
      <div class="font-xs my-2 font-bold">
        {{ $t('workTypeRequestModal.reason_requested') }}
      </div>
      <textarea
        v-model="reason"
        rows="4"
        class="block w-full border outline-none"
      />
    </div>

    <div slot="footer" class="p-3 flex justify-end">
      <base-button
        :text="$t('actions.cancel')"
        :alt="$t('actions.cancel')"
        class="ml-2 p-3 px-6 mr-1 text-xs border border-black"
        :action="
          () => {
            $emit('onCancel');
          }
        "
      />
      <base-button
        variant="solid"
        :action="
          () => {
            $emit('onRequest', {
              workTypes: Array.from(requestedWorkTypes),
              reason: reason,
            });
          }
        "
        :text="$t('actions.request')"
        :alt="$t('actions.request')"
        class="ml-2 p-3 px-6 text-xs"
      />
    </div>
  </modal>
</template>
<script>
import Organization from '@/models/Organization';

export default {
  name: 'WorkTypeRequestModal',
  props: {
    workTypes: {
      type: Array,
      default: () => {
        return [];
      },
    },
    initialSelection: {
      type: Array,
      default: () => {
        return [];
      },
    },
    myOrganization: {
      type: Object,
      default: () => {
        return {};
      },
    },
  },
  data() {
    return {
      requestedWorkTypes: new Set(this.initialSelection),
      reason: '',
      organizations: [],
    };
  },
  async mounted() {
    const organizationIds = this.workTypes.map((type) => type.claimed_by);
    this.organizations = Organization.query().whereIdIn(organizationIds).get();
  },
  methods: {
    setSelectedWorkTypeRequest(workType, value) {
      if (value) {
        this.requestedWorkTypes.add(workType);
      } else {
        this.requestedWorkTypes.remove(workType);
      }
    },
  },
};
</script>
