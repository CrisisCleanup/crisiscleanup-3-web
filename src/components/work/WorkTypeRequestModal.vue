<template>
  <modal
    :title="$t('workTypeRequestModal.work_type_request')"
    modal-classes="w-2/3"
    modal-body-classes="h-72 overflow-scroll border-b"
    @close="$emit('onCancel')"
  >
    <div class="px-6 py-3">
      <div class="font-xs my-2">
        <span
          v-html="
            $t('workTypeRequestModal.request_modal_instructions', {
              organizations: organizations.map((org) => org.name).join(','),
              my_organization: myOrganization.name,
              case_number: '',
            })
          "
        ></span>
        <span
          v-html="$t('workTypeRequestModal.please_add_respectful_note')"
        ></span>
        <ul class="my-2 list-disc list-inside">
          <li>
            {{ $t('workTypeRequestModal.reason_member_of_faith_community') }}
          </li>
          <li>
            {{ $t('workTypeRequestModal.reason_working_next_door') }}
          </li>
          <li>
            {{ $t('workTypeRequestModal.reason_we_did_the_work') }}
          </li>
        </ul>
      </div>
      <div class="font-xs my-2 font-bold">
        {{ $t('workTypeRequestModal.contacts') }}
      </div>
      <template v-for="organization in organizations">
        <div
          v-for="contact in organization.incident_primary_contacts"
          :key="contact.email"
        >
          {{ contact.first_name }} {{ contact.last_name }} ({{
            organization.name
          }}) {{ contact.email }} {{ contact.mobile }}
        </div>
      </template>
      <div class="font-xs my-2 font-bold">
        {{ $t('workTypeRequestModal.choose_work_types') }}
      </div>
      <div
        v-for="work_type_to_request in workTypes"
        :key="work_type_to_request.work_type"
      >
        <base-checkbox
          class="mb-3"
          :value="requestedWorkTypes.has(work_type_to_request.work_type)"
          @input="
            (value) => {
              setSelectedWorkTypeRequest(work_type_to_request.work_type, value);
            }
          "
        >
          <span class="text-sm">{{
            getWorkTypeName(work_type_to_request.work_type)
          }}</span>
        </base-checkbox>
      </div>
      <div class="font-xs my-2 font-bold">
        {{ $t('workTypeRequestModal.reason_requested') }}
      </div>
      <textarea
        v-model="reason"
        rows="4"
        class="block w-full border outline-none"
      />
    </div>
    <template #footer>
      <div class="p-3 flex justify-end">
        <base-button
          :text="$t('actions.cancel')"
          :alt="$t('actions.cancel')"
          class="ml-2 p-3 px-6 mr-1 text-xs border border-black"
          :action="
            () => {
              $emit('onCancel');
            }
          "
        />
        <base-button
          variant="solid"
          :action="
            () => {
              $emit('onRequest', {
                workTypes: Array.from(requestedWorkTypes),
                reason: reason,
              });
            }
          "
          :text="$t('actions.request')"
          :alt="$t('actions.request')"
          class="ml-2 p-3 px-6 text-xs"
        />
      </div>
    </template>
  </modal>
</template>
<script>
import Organization from '../../models/Organization';
import { onMounted, ref } from 'vue';
import { getWorkTypeName } from '../../filters/index';

export default {
  name: 'WorkTypeRequestModal',
  props: {
    workTypes: {
      type: Array,
      default: () => {
        return [];
      },
    },
    initialSelection: {
      type: Array,
      default: () => {
        return [];
      },
    },
    myOrganization: {
      type: Object,
      default: () => {
        return {};
      },
    },
  },
  setup(props) {
    const requestedWorkTypes = ref(new Set(props.initialSelection));
    const reason = ref('');
    const organizations = ref([]);

    function setSelectedWorkTypeRequest(workType, value) {
      if (value) {
        requestedWorkTypes.value.add(workType);
      } else {
        requestedWorkTypes.value.remove(workType);
      }
    }

    onMounted(() => {
      const organizationIds = props.workTypes.map((type) => type.claimed_by);
      organizations.value = Organization.query()
        .whereIdIn(organizationIds)
        .get();
    });

    return {
      requestedWorkTypes,
      reason,
      organizations,
      setSelectedWorkTypeRequest,
      getWorkTypeName,
    };
  },
};
</script>
