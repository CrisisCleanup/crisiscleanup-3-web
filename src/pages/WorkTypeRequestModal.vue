<template>
  <modal
    :title="$t('Work Type Request')"
    modal-classes="w-96"
    @close="$emit('onCancel')"
  >
    <div class="px-6 py-3">
      <div>
        {{
          $t('Choose work type you would like to claim for your organization')
        }}
      </div>
      <div v-for="work_type_to_request in work_types">
        <base-checkbox
          class="mb-5"
          :value="requestedWorkTypes.has(work_type_to_request.work_type)"
          @input="
            value => {
              setSelectedWorkTypeRequest(work_type_to_request.work_type, value);
            }
          "
        >
          <span class="text-sm">{{
            work_type_to_request.work_type | getWorkTypeName
          }}</span>
        </base-checkbox>
      </div>
    </div>

    <div slot="footer">
      <base-button
        type="primary"
        :action="
          () => {
            $emit('onRequest', Array.from(requestedWorkTypes));
          }
        "
        :text="$t('actions.request')"
        class="ml-2 p-1 px-3 text-xs"
      />
    </div>
  </modal>
</template>
<script>
export default {
  name: 'WorkTypeRequestModal',
  props: {
    work_types: {
      type: Array,
      default: () => [],
    },
    initialSelection: {
      type: Array,
      default: () => [],
    },
  },
  data() {
    return {
      requestedWorkTypes: new Set(this.initialSelection),
    };
  },
  methods: {
    setSelectedWorkTypeRequest(workType, value) {
      if (value) {
        this.requestedWorkTypes.add(workType);
      } else {
        this.requestedWorkTypes.remove(workType);
      }
    },
  },
};
</script>
