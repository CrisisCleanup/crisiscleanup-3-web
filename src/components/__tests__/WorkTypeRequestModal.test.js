/**
 *
 * Tests for WorkTypeRequestModal
 *
 * Components
 */

import BaseButton from '@/components/BaseButton';
import BaseCheckbox from '@/components/BaseCheckbox';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import { Store } from 'vuex-mock-store';
import WorkTypeRequestModal from '../WorkTypeRequestModal';

jest.mock('@/models/User');
jest.mock('@/models/Worksite');
jest.mock('@/models/Incident');
jest.mock('@/models/Organization');
jest.mock('@/models/WorksiteRequest');

const localVue = createLocalVue();
localVue.filter('getWorkTypeName', () => 'workType.ash');
localVue.use(Vuex);

const mocks = {
  $t: (key) => key,
  $store: new Store({}),
};

const mountWithOptions = (props) =>
  shallowMount(WorkTypeRequestModal, {
    stubs: {
      'base-button': BaseButton,
      'base-checkbox': BaseCheckbox,
      modal: true,
    },
    propsData: {
      workTypes: [],
      initialSelection: [],
      ...props,
    },
    store: mocks.$store,
    localVue,
    mocks,
  });

describe('WorkTypeRequestModal', () => {
  it('should not log any errors', () => {
    const spy = jest.spyOn(global.console, 'error');
    mountWithOptions({});
    expect(spy).not.toHaveBeenCalled();
  });

  it('show modal with options', () => {
    const wrapper = mountWithOptions({});
    expect(wrapper.element).toMatchSnapshot();
  });

  it('show modal with options', () => {
    const wrapper = mountWithOptions({
      workTypes: ['muck_out'],
    });
    expect(wrapper.vm.requestedWorkTypes).toMatchSnapshot();
  });

  it('show modal with options and initial selection', () => {
    const wrapper = mountWithOptions({
      workTypes: ['muck_out'],
      initialSelection: ['muck_out'],
    });
    expect(wrapper.vm.requestedWorkTypes).toMatchSnapshot();
  });

  it('show modal with options and multi initial selection', () => {
    const wrapper = mountWithOptions({
      workTypes: ['muck_out', 'ash', 'tarp'],
      initialSelection: ['muck_out', 'tarp'],
    });
    expect(wrapper.vm.requestedWorkTypes).toMatchSnapshot();
  });
});
