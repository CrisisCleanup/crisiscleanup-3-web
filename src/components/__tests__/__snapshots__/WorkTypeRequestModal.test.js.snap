// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkTypeRequestModal show modal with options 1`] = `
<modal-stub
  modal-classes="w-2/3"
  title="workTypeRequestModal.work_type_request"
>
  <div
    class="px-6 py-3"
  >
    <div
      class="font-xs my-2"
    >
      <span>
        workTypeRequestModal.request_modal_instructions
      </span>
       
      <span>
        workTypeRequestModal.please_add_respectful_note
      </span>
       
      <ul
        class="my-2 list-disc list-inside"
      >
        <li>
          
          workTypeRequestModal.reason_member_of_faith_community
        
        </li>
         
        <li>
          
          workTypeRequestModal.reason_working_next_door
        
        </li>
         
        <li>
          
          workTypeRequestModal.reason_we_did_the_work
        
        </li>
      </ul>
    </div>
     
    <div
      class="font-xs my-2 font-bold"
    >
      
      workTypeRequestModal.contacts
    
    </div>
      
    <div
      class="font-xs my-2 font-bold"
    >
      
      workTypeRequestModal.choose_work_types
    
    </div>
      
    <div
      class="font-xs my-2 font-bold"
    >
      
      workTypeRequestModal.reason_requested
    
    </div>
     
    <textarea
      class="block w-full border outline-none"
      rows="4"
    />
  </div>
   
  <div
    class="p-3 flex justify-end"
  >
    <button
      alt="button"
      class="ml-2 p-3 px-6 mr-1 text-xs border border-black flex items-center justify-center base-button undefined js-button"
      data-cy="js-button"
      title="Button"
      type="submit"
    >
      <!---->
       
      <!---->
       
      <!---->
       actions.cancel 
      <!---->
    </button>
     
    <button
      alt="button"
      class="ml-2 p-3 px-6 text-xs flex items-center justify-center base-button solid js-button"
      data-cy="js-button"
      title="Button"
      type="submit"
    >
      <!---->
       
      <!---->
       
      <!---->
       actions.request 
      <!---->
    </button>
  </div>
</modal-stub>
`;

exports[`WorkTypeRequestModal show modal with options 2`] = `Set {}`;

exports[`WorkTypeRequestModal show modal with options and initial selection 1`] = `
Set {
  "muck_out",
}
`;

exports[`WorkTypeRequestModal show modal with options and multi initial selection 1`] = `
Set {
  "muck_out",
  "tarp",
}
`;
